### PR TITLE
fix truncnorm with infinite mean

### PR DESCRIPTION
**Pull request by cossio**
fix https://github.com/JuliaStats/Distributions.jl/issues/624

### CODE REVIEW

**Comment by cossio** [comment]
The CI fails look unrelated to this PR.

**Comment by cossio** [comment]
Merge?

**Comment by cossio** [comment]
CC: @matbesancon

File: src/truncated/normal.jl [hunk 1]
```diff
@@ -141,10 +141,14 @@ function rand(rng::AbstractRNG, d::Truncated{Normal{T},Continuous}) where T <: R
     d0 = d.untruncated
     μ = mean(d0)
     σ = std(d0)
-    a = (d.lower - μ) / σ
-    b = (d.upper - μ) / σ
-    z = randnt(rng, a, b, d.tp)
-    return μ + σ * z
+    if isfinite(μ)
+        a = (d.lower - μ) / σ
```
Comment (matbesancon) [inline src/truncated/normal.jl:145]: if isfinite()
...
return ..
end
return end

no need for an else with early return

Comment (matbesancon) [reply src/truncated/normal.jl:145]: no that's alright by me, not worth bothering too much, the main concern is keeping the code homogeneous

**Pull request merged**